docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+- hello.hpp, the C++ header file.
+- hello.cpp, the C++ implementation file.
+
+#### Build the DUNE Bare Application
+
+To build the DUNE bare application run the following command:
+
+```shell
+dune -- cdt-cpp /host/<LOCAL-PATH-TO-HELLO-PARENT>/hello/hello.cpp -o /host/<LOCAL-PATH-TO-HELLO-PARENT>/hello/hello.wasm
+```
+
+The above command you can run it from directory on your local machine, just make sure that the `<LOCAL-PATH-TO-HELLO-PARENT>` is replace with the corresponding path of the parent directory of the hello folder.
+
+The result of the above build command are two files:
+
+- hello.wasm, the WebAssembly binary file for the smart contract.
+- hello.abi, the application binary interface (ABI) file for the smart contract.
+
+#### Deploy the hello Smart Contract (built with cdt-cpp)
+
+Execute the following command to deploy the `hello` smart contract to the account `hello` which you created at the beginning:
+
+```shell
+dune --deploy <PATH_TO_HELLO_DIR>
+```
+
+### Create, Build and Deploy a DUNE cmake Application
+
+The DUNE cmake application is a more complex type than the bare type and is suited better for larger projects which involve multiple contracts and/or web applications.
+
+```shell
+dune --create-cmake-app hello .
+cd hello
+ls
+```
+
+The result of this command is a `hello` directory with a bit more complex structure than the bare application because it generates some extra files to configure the cmake project:
+
+- CMakeLists.txt, cmake configuration file.
+- README.txt, a text file which contains information about how to build this application with cmake.
+- build, the output build folder, at the beginning is empty.
+- include, the C++ include files folder, at the beginning it contains just hello.hpp file.
+- ricardian, the folder contains the smart contract ricardian definition, hello.contracts.md file.
+- src, the C++ implementation files folder, at the beginning it contains just hello.cpp file.
+
+#### Build the DUNE cmake Application
+
+To build the DUNE cmake application run the following command:
+
+```shell
+dune --cmake-build <PATH_TO_CMakeLists.txt_PARENT_DIR>
+ls <PATH_TO_CMakeLists.txt_PARENT_DIR>/build/hello
+```
+
+You can run the two commands from above from any directory on your local machine, just make sure you correctly replace the `<PATH_TO_CMakeLists.txt_PARENT_DIR>` with the parent directory path of `CMakeLists.txt` file.
+
+The result of the above build command are two files located in the `./build/hello/` folder:
+
+- hello.wasm, the WebAssembly binary file for the smart contract.
+- hello.abi, the application binary interface (ABI) file for the smart contract.
+
+#### Deploy the hello Smart Contract (built with cmake)
+
+Execute the following command to deploy the `hello` smart contract to the account `hello` which you created at the beginning:
+
+```shell
+dune --deploy <PATH_TO_CMakeLists.txt_PARENT_DIR>/build/hello
+```
+
+## First Test
```

Comment:
I'm not sure if this section really adds anything to the dev's journey. They'll be running actions later anyway.

Reply:
when I start a guide, tutorial which generates a boiler plate code for me, I wanna build, deploy and run to see how it works un-modified, at the beginning,  bare bones.

Reply:
ok I took it out